internal/resources/factories.ts

```diff
@@ -142,7 +142,7 @@ export function createResourceFromServer(server: net.Server, name: string = "net
     finalize: () => {
       return new Promise((resolve, reject) => {
         server.close((err) => {
-          if (err === undefined) {
+          if (err !== undefined) {
```

Comment:
Actually shouldn't this be:

```suggestion
          if (err != null) {
```